tsdb/compact.go

```diff
@@ -521,7 +525,7 @@ func (c *LeveledCompactor) CompactWithBlockPopulator(dest string, dirs []string,
 				"duration", time.Since(start),
 			)
 		}
-		return uid, nil
+		return []ulid.ULID{uid}, nil
```

Comment:
We need to check if uid is empty and return an empty slice here. Since this was not caught by the tests, we need to test this. We can have a block with 1 sample and tombstone covering entire block. Then calling Compact() on this block should return empty slice.

Reply:
Make sense! Added a unit test case `TestCompactEmptyResultBlockWithTombstone`.